Add Anisotropic Filtering and Wrap Options to gltf Reimport.

Fristender opened the pull request:
<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->
When importing a gltf asset (sponza tested), the default configuration uses linear filtering for minification/magnification which makes viewing from oblique angles blurry. This change adds the option for the user to decide which filtering/wrap method they want for imported gltfs, including linear and nearest anisotropic filtering to reduce blur.

Fristender committed:
Add filtering to import options.

Fristender committed:
Add anistropic filtering and correctly apply filtering and wrap for gltf imports.

Fristender committed:
Fix set_wrap_t bug.

Fristender committed:
Fix Minification default display bug and Linux workflow error.

Fristender committed:
Fix make_rst error.

Fristender committed:
Move enum binding to cpp.

Fristender committed:
Trigger GHA workflow.

Fristender committed:
Revert "Trigger GHA workflow."

This reverts commit 74c39bb918950c0c1407358e0ec4ec74d13982b2.

Fristender committed:
Attempt to fix with --doctool.

aaronfranke commented on modules/gltf/gltf_document.cpp:
```diff
@@ -4464,7 +4464,26 @@ Error GLTFDocument::_parse_texture_samplers(Ref<GLTFState> p_state) {
+				}
+			}
+		}
+		sampler->set_min_filter(p_state->get_texture_filter_min());
```
This PR doesn't make any sense for multiple reasons:

* You are overriding the values set in the above `if` statements always, making the `if` statements useless.
* These are settings already specified each texture sampler, so there is no need for these settings in GLTFState.
* The code in GLTFState and GLTFTextureSampler is, for the most part, meant to represent data in the glTF file. If the intention is to override what is specified in the glTF file, that should be done in ResourceImporterScene.

Fristender committed:
Adds new import options to the scene importer for overriding texture filtering and repeat modes on materials within the imported scene. This functionality was previously part of a glTF extension, but has been moved into the core scene importer to support other formats like FBX.

The implementation includes:

    -New texture options in ResourceImporterScene.

    -Removal of the old glTF-specific options and state.

    -A refactored _apply_texture_filter_to_materials function to correctly apply the new options.

    -glTF import defaults to what it used to be (linear + repeat)

Fristender committed:
Update with --doctool.
